spark/src/main/resources/python/zeppelin_pyspark.py

```diff
@@ -44,15 +49,59 @@ def flush(self):
 class PyZeppelinContext(dict):
   def __init__(self, zc):
     self.z = zc
+    self.max_result = 1000
     self._displayhook = lambda *args: None
 
+  def _isPandas(self, obj):
+    try:
+      from pandas.core.frame import DataFrame
+      return isinstance(obj, DataFrame)
+    except ImportError:
+      return false
+
   def show(self, obj):
     from pyspark.sql import DataFrame
     if isinstance(obj, DataFrame):
       print(self.z.showData(obj._jdf))
+    elif self._isPandas(obj):
+      self.show_dataframe(obj)
     else:
       print(str(obj))
 
+  def show_dataframe(self, df, show_index=False):
+    """Pretty prints DF using Table Display System
+    """
+    limit = len(df) > self.max_result
+    header_buf = StringIO("")
+    if show_index:
+      idx_name = str(df.index.name) if df.index.name is not None else ""
+      header_buf.write(idx_name + "\t")
+    header_buf.write(str(df.columns[0]))
+    for col in df.columns[1:]:
+      header_buf.write("\t")
+      header_buf.write(str(col))
+    header_buf.write("\n")
+
+    body_buf = StringIO("")
+    rows = df.head(self.max_result).values if limit else df.values
+    index = df.index.values
+    for idx, row in zip(index, rows):
+      if show_index:
+        body_buf.write("%html <strong>{}</strong>".format(idx))
+        body_buf.write("\t")
+      body_buf.write(str(row[0]))
+      for cell in row[1:]:
+        body_buf.write("\t")
+        body_buf.write(str(cell))
+      body_buf.write("\n")
+    body_buf.seek(0); header_buf.seek(0)
+    #TODO(bzz): fix it, so it shows red notice, as in Spark
```

Comment:
Actually all the code of method `show_dataframe` are copied from python interpreter, I don't have much context about this piece of code.  Ideally we should have all the logic in python interpreter, and `PySparkInterprete` rshould just reuse or extend python interpreter. But this would be a large follow up ticket.